DeAceroDLabs/marketplace-frontend
# Feature/dmp 604 github actions for deploy

## What is done

[[DMP-604] Github actions frontend](https://deacero.atlassian.net/browse/DMP-604)

## How is done

- Added a deploy workflow that is used on main.
- Added secret actions so that will be read on variables main.yml.
- Added deployer.pem on a folder locally on my computer...
- developer.pem contains:
```
-----BEGIN PRIVATE KEY-----
MIIEvQIBADANBgkqhkiG9w0BAQEFAASCBKcwggSjAgEAAoIBAQCdMmhSy8spxVv/
fqNAhK0cnLz82dBzYIbDrYI/NOCIH0GcHpdGeat26PWoIEw53JpMhNiROs+C84Lv
Xw6HxkOTPM1qr58lmJBEP5rJszGIMt1i60sGnrlpR+OODe7QlDc2BfcM/cNgXqFJ
uq+1Y/hLsJHOwUSgxNHcND/r9daOhgh02K16w7N6QwEM0AUBa2fMHXEt9AlaQS5N
iLO55+lFNa8aXNekxViE2mi3EaZ3AtAj34O8Xwg9D4el6NkJtvQoJfG4SBFtxz1J
YFi2Yrc3Iw9ICSrFCajXjZWAZ1oMdLAr8Rsg4ZEzGt/EP62KdoAy9YCf2kQd15Vc
MMO2cnwXAgMBAAECggEAOqXq3DEifFQgzcLun4yglZM4w1F0fwddYDYk6PYOa1rV
/KxBHk8YlMW3mWFrbw8KFlBeCFkbwKyfhjU905Q58/30WcBqLPmtoXjlhCkx6rGn
fVdx52din4/tqzsxhWo26xXwpU0sDpFftYVMNu20XnDjBXN/6kdkULbBvoQ653Mu
ljrYNTV4EsH/pIz5ihgZI2VGXSaeFaiRxg2gK9rp9oDSBa2CLM6b4VOtDDbuWV69
3Tok2lJHkhmVPb258SGiPTy9KEXMks7RWyuW1WXhRg5SUc/9nYRxLy7sJBFQmM0m
IsR2gECc8bbVe7yoOR+TFtJZGIzLK+Xy5W5Z5t0iQQKBgQDIm+6b+sxBE0p+kSXa
khIwDPeJxxO9FdhfOxhfE/c9e6gHni3xHuQuVEfRAlthzH6+bI2HaFjY4DeHljo9
WpxdgCHIKEiU6IHHvq2hicMDU+AgJyP9xr0LQ6jdkZJ3p7TFL1UOZLhAqS3DNwwx
GHIyAvGMcEshDSb8w4OYTLFHIQKBgQDImeGChae3KpGrgkkj2++S5zDxFQMWRV2k
tvV8GAT8Dkpmg5chTDwD/FUWn5gpS2f8hnpeB1SjIpO46i3c5GPE9KVVCJHuR8cD
IGskozAuOA0VV6DzBB61Q7wPdik2bIVhmfH76a7zWmJ3f26EFIsig2y6KZ0F44V2
UFutwDm0NwKBgDOxuvMG1cuPJQsobBC+w2EOdltSqYt4LaGcoNMfWf7F9q6/3Cbe
Rv6JSVhauyGgiKiZYH8cLIj2fli6JZfZ2aFee1YQy2NT5t2KDUQrM82Pt5mRfHzQ
0bYtYT53XL7owHRvlgN3gONotRow/TtJiWPJzmTaKwIBWaKY3zhHywcBAoGBAJHl
oWddsBJ+6n2T/4AOKRtt/GGLB7i98JKo4rhDe/SarDANCvET0h6P+o6+ultmVZYL
dJpIuJ0Bu/DkmUCr94g66riCyChGmg7DGEoISjWDXLimCBBXccIiJngUV3sWyJeo
8J9oYW7q0OzaiNWVX3BVp9FqDKNIzDu1zEf9o7pXAoGAEF2l30bKhbi9pNqlFjkZ
NFENzW3ej6F45HYdbo+DwJrTHJmPBtHQDZJ2vvWjVeskKLP3jVZr8rAugEcgEJcn
nQckoAiOZkK/TmkBEbUDxDO989llwAzB1QTKfFiBnkysWVQM+7YT3/DUtI6P+IAJ
AygX93xtfs8g/9OPruMnEQc=
-----END PRIVATE KEY-----
```
- Used command: ` sudo ssh -i deployer.pem deployer@ec2-18-116-181-37.us-east-2.compute.amazonaws.com`
- And lastly created mkt-place-front.env file that contains the env variables that are needed to run the proyect.
![image](https://user-images.githubusercontent.com/81311646/195698043-a87a5d02-2fce-4f32-bb55-1135756f3e14.png)

- Added also all the secret actions on the repo, you can see them on: settings>secret>actions
- These secrets are only given directly from an aws admin: Manuel Becerra or Erving Uriel Castillo Ramos. Specially for prod.
![image](https://user-images.githubusercontent.com/81311646/195705529-4eaf0dcc-ed2e-4e77-8a41-6ca236e88949.png)

## How to test

1. Please check first: https://github.com/DeAceroDLabs/mkt-place-back/pull/18
2. This would be test after being merged, cause it is the way it works.
3. But can check the code.
4. In case anything goes wrong, there would be a fix.
